exec/cpg.c

```diff
@@ -734,11 +741,13 @@ static int notify_lib_joinlist(
 	qb_list_for_each(iter, &cpg_pd_list_head) {
 			struct cpg_pd *cpd = qb_list_entry (iter, struct cpg_pd, list);
 			if (mar_name_compare (&cpd->group_name, group_name) == 0) {
-				assert (joined_list_entries <= 1);
```

Comment:
there was a similar assert for the left_list_entries which was removed in the referenced commit. both were originally introduced in 058f5031 "Rewrite of CPG. It solves problems with double delete." in 2009..

Reply:
Yep, delete of this assert is ok. It was there only to ensure that only 1 item is in the list (I'm wondering why `left_list_entries` was deleted when it reads only first item....)

Reply:
Studying code a little bit more I think removal of `left_list_entries` is actually really a bug and it works only because of:
- `downlist_inform_clients` can send more than one left entry, but it never contains current node in left list so code is no-op
- `do_proc_leave` - just one left entry